### PR TITLE
spawn recipes on startup accorging to @trigger annotations

### PR DESCRIPTION
**Pull request by mmandlis**
implements: https://github.com/PolymerLabs/arcs/issues/3823

### CODE REVIEW

File: javaharness/java/arcs/android/ArcsService.java [hunk 1]
```diff
@@ -22,12 +25,23 @@
...
     DaggerArcsServiceComponent.builder().build().inject(this);
     arcsShellApi.init(this);
+    environment.addReadyListener(recipes -> {
+      recipes.forEach(recipe -> {
+        ArcData arcData = findStartupTrigger(recipe);
+        if (arcData != null) {
+          arcsShellApi.startArc(arcData);
+        }
+      });
+    });
```
Comment (yuangu-google) [inline javaharness/java/arcs/android/ArcsService.java]: Please move this part to ArcsShellApi which wraps ArcsEnvironment. ArcsShellApi is the thing wraps the real shell api, which does the actual start up.

File: javaharness/java/arcs/android/ArcsService.java [hunk 2]
```diff
@@ -22,12 +25,23 @@
   @Inject
   ArcsShellApi arcsShellApi;
 
+  @Inject
+  AndroidArcsEnvironment environment;
```
Comment (yuangu-google) [inline javaharness/java/arcs/android/ArcsService.java]: This new field can be avoided if we move the logic to ArcsShellApi.

File: javaharness/java/arcs/android/ArcsService.java (hunk 1 above)
Comment (mmandlis) [reply javaharness/java/arcs/android/ArcsService.java]: done

File: javaharness/java/arcs/android/ArcsService.java (hunk 2 above)
Comment (mmandlis) [reply javaharness/java/arcs/android/ArcsService.java]: done

**Comment by mmandlis** [comment]
addressed comments, PTAL

File: shells/pipes-shell/source/pipe.js [hunk 3]
```diff
@@ -50,17 +48,15 @@ export const initPipe = async (client, paths, storage) => {
 
 // TODO(sjmiles): must be called only after `window.ShellApi` is initialized
 export const initArcs = async (storage, bus) => {
-  // marshal ingestion arc
```
Comment (mmandlis) [inline shells/pipes-shell/source/pipe.js]: @sjmiles Scott, please, hold off reviewing. 
sorry about this, I remember now that I shouldn't be just removing this - I'll send a commit shortly to restore the sharing functionality.

Comment (mmandlis) [reply shells/pipes-shell/source/pipe.js]: updated the PR, PTAL

**Pull request merged**